src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -148,10 +149,9 @@
                         "jruby-puppet.max-active-instances.")}))
      (when (jruby-schemas/shutdown-poison-pill? pool-instance#)
        (jruby/return-instance ~jruby-service pool-instance# ~reason)
-       (sling/throw+
-        {:type    ::service-unavailable
-         :message (str "Attempted to borrow a JRuby instance from the pool "
-                       "during a shutdown. Please try again.")}))
+       (mw/throw-service-unavailable!
```

Comment:
Yeah, after all that, it would probably have been better to at least break out the helpers into a different namespace so downstream consumers could do something like:
```clj
(ns puppetlabs.example.test
  (:require [puppetlabs.ring-middleware.core :as middleware]
            [puppetlabs.ring-middleware.utils :as ringutils]))

(let [stack (middleware/wrap-bad-request
              (fn [_] (ringutils/throw-bad-request! "Example")))
      response (stack (basic-request)]
  (is (logged? #".*Example.*" :error)))
```

It might also be a good idea because I would guess, outside of tests, consumers would use the different functionality in different namespaces as well.